app/handlers/profiles/settings/avatar.go

```diff
@@ -44,20 +43,40 @@ func AvatarSubmitHandler(c *gin.Context) {
 	}
 	img = resize.Thumbnail(256, 256, img, resize.Bilinear)
 
-	f, err := os.Create(fmt.Sprintf("%s/%d.png", settings.APP_AVATAR_PATH, ctx.User.ID))
-	defer f.Close()
+	f, err := os.CreateTemp("", fmt.Sprintf("%d.png", ctx.User.ID))
 	if err != nil {
 		m = msg.ErrorMessage{lu.T(c, "An error occurred.")}
 		c.Error(err)
 		return
 	}
+	defer os.Remove(f.Name())
 
 	err = png.Encode(f, img)
 	if err != nil {
 		m = msg.ErrorMessage{lu.T(c, "We were not able to save your avatar.")}
 		c.Error(err)
 		return
 	}
+	// seek file to beginning
+	f.Seek(0, os.SEEK_SET)
+
+	sess := session.Must(session.NewSession(&aws.Config{
+		Region:   aws.String(settings.AWS_REGION),
+		Endpoint: aws.String(settings.AWS_ENDPOINT_URL),
+	}))
+	uploader := s3manager.NewUploader(sess)
+	result, err := uploader.Upload(&s3manager.UploadInput{
+		Bucket:      aws.String(settings.AWS_BUCKET_NAME),
+		Key:         aws.String(fmt.Sprintf("avatars/%d.png", ctx.User.ID)),
+		Body:        f,
+		ContentType: "image/png",
+		// TODO: CacheControl?
+	})
+	if err != nil {
+		m = msg.ErrorMessage{lu.T(c, "We were not able to save your avatar.")}
+		c.Error(err)
+		return
+	}
```

Comment:
allegedly credentials should work directly from env: https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#:~:text=profile%20you%20specify.-,Environment%20Variables,-By%20default%2C%20the